py-polars/tests/unit/datatypes/test_decimal.py

```diff
@@ -84,6 +84,15 @@ def test_init_decimal_dtype() -> None:
     assert df["a"].is_numeric()
 
 
+def test_decimal_cast_by_schema(float_tolerance: float = 0.01) -> None:
+    df = pl.DataFrame(
+        {"a": [D("1"), D("2.55"), D("45.000"), D("10.0")]}, schema={"a": pl.Float64}
+    )
+    assert (
+        abs(df["a"].sum() - 58.55) < float_tolerance
+    ), "Something went wrong converting decimals to floats"
```

Comment:
We don't really need assert messages, as the test name already documents this for us.

I would write the assertion like this:

```suggestion
    expected = pl.DataFrame({"a": [1.0, 2.55, 45.0, 10.0]})
    assert_frame_equal(df, expected)
```

Our assertion utils have float tolerance built-in.

Reply:
Thanks @stinodego !

Reply:
I'm not sure what caused the original bug, but your test still doesn't properly test for it. Your horizontal example also passes on `main`.

I'd say always add the minimal reproducible example from the original issue as a test, and we often include the issue number in the test name as well.

Reply:
I went ahead and added the test myself!